### DISCUSSION TITLE
time step selection problem

### DISCUSSION THREAD

**Discussion by hekungeo** [Anura3D/Anura3D_OpenSource#17] [Q&A] [answered]
Hi Dear all,

I am doing research on the antecedent rainfall-induced landslide. The previous precipitation lasted for 15 days before the slope failure. How can I determine the time step in GID, because the unit is second, selecting the one day (86400s) for one-time step seems to require lots of time and may cause unstable calculation. Or can I simplify the 15 days to 15 seconds by converting the infiltration rate? such as 70 mm/day to 8e-7 m/s. 

Thx in advance.

Best,
Kun

**Comment by murphy-john** [comment, marked as answer]
Hi Kun,
I think you mean the "load step" or "calculation step" that's defined in GiD and written into the CPS_001 file. The "time step" is a subset of the time in the "load step" and is calculated in the code based on the CFL condition and the $$COURANT_NUMBER in the CPS_001. The stability of the calculation shouldn't depend on the time you put in for the Time per Load/Calculation step. 

I'm not aware of anyone who's run models with Anura3D that have lasted for 15 days (model time). The longest models I've run have been for 50 hrs (model time). There are ways to reduce the run time of the code, as you've mentioned. It's up to the modeler to determine the best course for their problem.

**Reply by LongTIme1111** [reply]
hi I have the same problem as him, that is, it takes a long time to simulate a rainfall landslide, so what is the specific way to reduce the calculation time?
